src/iterative_ensemble_smoother/experimental_sies.py

```diff
@@ -0,0 +1,315 @@
+from __future__ import annotations
+from typing import Optional, TYPE_CHECKING, Callable
+
+import numpy as np
+import scipy as sp
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+
+
+# =============================================================================
+# MAIN CLASS
+# =============================================================================
+
+
+class SIES:
+    def __init__(
+        self,
+        param_ensemble: npt.NDArray[np.double],
+        observation_errors: npt.NDArray[np.double],
+        observation_values: npt.NDArray[np.double],
+        *,
+        seed: Optional[int] = None,
```

Comment:
Here I want to deviate a little bit from the paper.

**I think we can safely assume it's not.** If it was, then we would just use standard results for posteriors of multivariate normal distributions. Newtons steps and ensemble smoothers would be overkill for linear models anyway.

Even if it is linear, it has no cost except an unnecessary projection since $A_i^+ A_i = I$ in the linear case anyway.